### PR TITLE
Parachains hrmp.rs to Frame V2

### PR DESCRIPTION
**Pull request by ferrell-code**
relates: #2882 

Following the upgrade guidelines here: https://crates.parity.io/frame_support/attr.pallet.html#upgrade-guidelines.

# :warning: Breaking Change :warning: 
From https://crates.parity.io/frame_support/attr.pallet.html#checking-upgrade-guidelines

 >storages now use PalletInfo for module_prefix instead of the one given to decl_storage: Thus any use of this pallet in construct_runtime! should be careful to update name in order not to break storage or to upgrade storage (moreover for instantiable pallet). If pallet is published, make sure to warn about this breaking change.

So users of the `Hrmp` pallet must be careful about the name they used in `construct_runtime!.` Hence the runtime-migration label, which might not be needed depending on the configuration of the `Hrmp` pallet.

rococo uses `Hrmp` in `construct_runtime!`,  so no mirgration needed

kusama and westend use `ParasHrmp` in `construct_runtime!` so I migrated them to `Hrmp`

polkadot does not us `Hrmp`.

### CODE REVIEW

**Comment by rphmeier** [comment]
Sorry for the delay on reviewing these.

**Comment by ferrell-code** [comment]
No worries, I'm chillin

**Review by shawntabrizi — APPROVED** [review]

File: runtime/parachains/src/hrmp.rs [hunk 1]
```diff
@@ -215,139 +215,46 @@ impl fmt::Debug for OutboundHrmpAcceptanceErr {
...
-#[cfg(feature = "std")]
-fn initialize_storage<T: Config>(preopen_hrmp_channels: &[(ParaId, ParaId, u32, u32)]) {
-	let host_config = configuration::Module::<T>::config();
-	for &(sender, recipient, max_capacity, max_message_size) in preopen_hrmp_channels {
-		if let Err(err) = preopen_hrmp_channel::<T>(sender, recipient, max_capacity, max_message_size) {
-			panic!("failed to initialize the genesis storage: {:?}", err);
-		}
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
```
Comment (thiolliere) [inline runtime/parachains/src/hrmp.rs:246]: I think event doesn't needs to be generic, however I'm ok with it anyway.

**Review by thiolliere — APPROVED** [review]

**Comment by thiolliere** [comment]
bot merge

**Pull request merged**